cmd/list/oidcprovider/cmd.go

```diff
@@ -113,21 +126,41 @@ func run(cmd *cobra.Command, _ []string) {
 		os.Exit(0)
 	}
 
+	filterProviders := false
+	if cmd.Flags().Changed("oidc-config-id") {
+		filterProviders = true
+	}
+
 	// Create the writer that will be used to print the tabulated results:
 	writer := tabwriter.NewWriter(os.Stdout, 0, 0, 2, ' ', 0)
 	fmt.Fprintf(writer, "OIDC PROVIDER ARN\tCluster ID\tIn Use\n")
 	for _, provider := range providers {
-		providerInUse := "No"
-		if ok := providersInUse[provider.Arn]; ok {
-			providerInUse = "Yes"
+		if !filterProviders {
+			printProvider(writer, providersInUse, provider)
+		} else {
```

Comment:
@hunterkepley How about we extend or add another function to `AWSClient` that allows you to provide a filter when retrieving the list of OIDC providers? We can then move the filtering logic there rather than in the command implementation and we continue to have Cobra as a thin wrapper around our APIs.

It would also make it easier for you to write a unit test for the filtering logic :-)